wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -730,44 +740,144 @@ public void init(InitRequest request) throws GitException {
+                                     .setListMode(mode)
+                                     .setContains(commit.getName())
+                                     .call();
+        return branches.stream()
+                       .map(branch -> newDto(Branch.class).withName(branch.getName()))
+                       .collect(Collectors.toList());
+    }
+
+    private List<DiffCommitFile> getCommitDiffFiles(RevCommit revCommit, String pattern) throws IOException {
+        List<DiffEntry> diffs = null;
+        TreeFilter filter = null;
+        if (pattern != null && !pattern.isEmpty()) {
+            filter = AndTreeFilter.create(PathFilterGroup.createFromStrings(Collections.singleton(pattern)), TreeFilter.ANY_DIFF);
+        }
+        List<DiffCommitFile> commitFilesList = new ArrayList<>();
+        try (TreeWalk tw = new TreeWalk(repository)){
+            tw.setRecursive(true);
+            // get the current commit parent in order to compare it with the current commit
+            // and to get the list of DiffEntry.
+            if (revCommit.getParentCount() > 0) {
+                RevCommit parent = parseCommit(revCommit.getParent(0));
+                tw.reset(parent.getTree(), revCommit.getTree());
+                if (filter != null) {
+                    tw.setFilter(filter);
+                } else {
+                    tw.setFilter(TreeFilter.ANY_DIFF);
+                }
+                diffs = DiffEntry.scan(tw);
+            } else {
+                // If the current commit has no parents (which means it is the initial commit),
+                // then create an empty tree and compare it to the current commit to get the
+                // list of DiffEntry.
+                RevWalk rw = null;
+                DiffFormatter diffFormat = null;
+                try {
+                    rw = new RevWalk(repository);
+                    diffFormat = new DiffFormatter(NullOutputStream.INSTANCE);
+                    diffFormat.setRepository(repository);
+                    if (filter != null) {
+                        diffFormat.setPathFilter(filter);
+                    }
+                    diffs = diffFormat.scan(new EmptyTreeIterator(), new CanonicalTreeParser(null, rw.getObjectReader(), revCommit.getTree()));
+                } finally {
+                    if (diffFormat != null) {
+                        diffFormat.close();
+                    }
+                    if (rw != null) {
+                        rw.close();
+                    }
+                }
+            }
+        }
+        if (diffs != null) {
+            commitFilesList.addAll(diffs.stream().map(diff -> newDto(DiffCommitFile.class).withOldPath(diff.getOldPath())
+                                                                                             .withNewPath(diff.getNewPath())
+                                                                                             .withChangeType(diff.getChangeType().name()))
```

Comment:
Please format this according to https://github.com/codenvy/codenvy-dev-resources/wiki/Formatting#for-dot--separator-break-always-comes-before-the-symbol